configure.ac

```diff
@@ -248,17 +248,24 @@ OPAL_CHECK_OFI([ofi],
     [transport_ofi="yes"],
     [transport_ofi="no"])
 
-# assure that only ofi is enabled if portals4 is also detected
-if test -n "$with_ofi" -a "$with_ofi" != "no" ; then
+# If both OFI and portals4 requested, user needs to choose one:
+if test -n "$with_ofi" -a "$with_ofi" != "no" -a -n "$with_portals4" -a "$with_portals4" != "no" ; then
+    AC_MSG_ERROR([Cannot choose both OFI and portals4 transports, use --with to choose one])
+# Check which was requested, OFI or portals4:
+elif test -n "$with_ofi" -a "$with_ofi" != "no" ; then
     transport="ofi"
     transport_portals4="no"
     AC_DEFINE([USE_OFI], [1], [Define if OFI transport active])
 elif test -n "$with_portals4" -a "$with_portals4" != "no" ; then
     transport="portals4"
     transport_ofi="no"
     AC_DEFINE([USE_PORTALS4], [1], [Define if Portals 4 transport active])
+# If neither, set transport to "none" and print a warning:
 else
-    AC_MSG_WARN([Neither OFI nor portals4 transport found])
+    transport="none"
+    transport_ofi="no"
+    transport_portals4="no"
+    AC_MSG_WARN([Neither OFI nor portals4 transport requested])
```

Comment:
I believe the final check prints only when no transports at all are found.  For example, if XPMEM is enabled but neither OFI nor Portals, that warning will not print.  This separate warning has been helpful for me, but I can remove it if you want.